### PR TITLE
changes to not use headers #3071

### PR DESCRIPTION
**Pull request by myleshorton**

### CODE REVIEW

**Comment by myleshorton** [comment]
Here are the enproxy changes @aranhoide

File: src/github.com/getlantern/enproxy/proxy.go [hunk 1]
```diff
@@ -128,6 +129,37 @@ func (p *Proxy) Serve(l net.Listener) error {
 	return httpServer.Serve(l)
 }
 
+func (p *Proxy) parseRequestProps(resp http.ResponseWriter, req *http.Request) (string, string, string, error) {
+	path := req.URL.Path
+
+	// If it's a reasonably long path, it likely follows our new request URI format:
+	// /X-Enproxy-Id/X-Enproxy-Dest-Addr/X-Enproxy-Op
+	if len(path) > 5 {
+		r, err := regexp.Compile("/(.*)/(.*)/(.*)")
```
Comment (fffw) [inline src/github.com/getlantern/enproxy/proxy.go]: Try `"^/(.*)/(.*)/(.*)"`? This regex matches the extra trailing slash.

> Sep  1 04:10:14 fl-nl-ara-001 flashlight: Sep 01 04:10:14.939 - 0m58s DEBUG enproxy: proxy.go:134 Path is /1f084d96-fe6b-4920-a139-d950ae54d759/twitter.com:443/write/

> Sep  1 04:10:14 fl-nl-ara-001 flashlight: Sep 01 04:10:14.940 - 0m58s DEBUG enproxy: proxy.go:185 Parsed enproxy data id: 1f084d96-fe6b-4920-a139-d950ae54d759/twitter.com:443, addr: write, op:

Comment (fffw) [inline src/github.com/getlantern/enproxy/proxy.go]: Oh wrong point sorry. It could be `/([^/]+)/([^/]+)/([^/]+)`

Comment (myleshorton) [inline src/github.com/getlantern/enproxy/proxy.go]: Hey yeah I was just fixing that too -- good catch! I did a little refactoring too.

**Comment by myleshorton** [comment]
This seems like it's now working for me @fffw. How about you?

**Comment by fffw** [comment]
Cool @myleshorton! It works steadily. A bit slower than chained server
which is expected.

File: src/github.com/getlantern/enproxy/proxy.go [hunk 2]
```diff
@@ -128,6 +129,40 @@ func (p *Proxy) Serve(l net.Listener) error {
 	return httpServer.Serve(l)
 }
 
+func (p *Proxy) parseRequestPath(path string) (string, string, string, error) {
+	log.Debugf("Path is %v", path)
+	r, err := regexp.Compile("/(.*)/(.*)/(.*)/")
```
Comment (fffw) [inline src/github.com/getlantern/enproxy/proxy.go]: I would use `regexp.MustCompile` and save it as a `var`

Comment (myleshorton) [inline src/github.com/getlantern/enproxy/proxy.go]: Agreed -- will make that change.

Comment (myleshorton) [inline src/github.com/getlantern/enproxy/proxy.go]: OK done!

**Comment by fffw** [comment]
Already updated server side in production. Test is good, merging. Great finding and fix @myleshorton !

**Pull request merged**